### ISSUE TITLE
Components lose direction when removed/re-attached to DOM

### ISSUE THREAD

**Issue by adixon-adobe** [adobe/spectrum-web-components#894]
### Expected Behaviour
If you remove a component from the DOM and re-add, it should still have a direction attribute (appropriately inherited).

### Actual Behaviour
The direction attribute doesn't get re-added properly when the element is re-attached.

#### Steps to Reproduce
1. Go to the https://opensource.adobe.com/spectrum-web-components/components/checkbox
2. Inspect a checkbox and choose "store as global variable" in chrome.
3. Paste the following into the chrome console:
```js
parent = temp1.parentElement
parent.removeChild(temp1)
parent.appendChild(temp1)
```

Result: there's no longer any padding between the checkbox & label, and the direction attribute is no longer on the checkbox.

**Comment by adixon-adobe** [comment]
Here's a screenshot: <img width="1003" alt="Screen Shot 2020-09-22 at 11 07 40 AM" src="https://user-images.githubusercontent.com/48841507/93908102-d7595680-fcc3-11ea-8fa9-7337641c7856.png">

**Comment by Westbrook** [comment]
On first glance, this seems to work _in RTL_, so hopefully it's just a little bit of logic somewhere that's missing?

**Comment by Westbrook** [comment]
It's right here:

https://github.com/adobe/spectrum-web-components/blob/5ebc7cf7dc50e8d8a95430eddb572d439d95f0e9/packages/base/src/Base.ts#L87

I remember specifically pointing to `this.dir` at the end, but I can't remember why... It won't be set if you get into this logic, so it'll apply `null`. We can move it back to `ltr` or to `this.dir || 'ltr'` and the issue should go away.

**Comment by adixon-adobe** [comment]
I'd *guess* that you have `this.dir` for cases where there's an explicit override, but that you want `this.dir || 'ltr'` to cover the more common case where it's not.

**Comment by Westbrook** [comment]
That's what I thought (looking at the code just now), but you can't get into this logic if `this.hasAttribute('dir')` have a value, which means that _always_ `this.dir` will be `null` 🙈

This may be a case of me _thinking_ "I'll do a smart thing here...", and it not actually being all the smart. Sadly all too common.

**Comment by Westbrook** [comment]
fixed by #895
